protocol/smart-contract-versioning.md

```diff
@@ -0,0 +1,141 @@
+- Subsequent changes only bump the `rc.n` identifier at the monorepo release level. One exception: a contract change results in a higher version level than prior bumps, e.g. if `rc.1` only bumped the minor version, but now we must introduce a breaking change for `rc.2`, we need to bump the major version. Ideally once we're at an `rc.n` this should not occur. This also results in a new branch being created.
+- Once the governance proposal passes, create the `op-contracts/vX.Y.Z` tag and release from the latest `op-contracts/vX.Y.Z-rc.n` release, then merge the branch into develop.
+
+## Monorepo Contracts Release Versioning
+
+Versioning for monorepo releases works as follows:
+
+- Monorepo releases continue to follow the `op-contracts/vX.Y.Z` naming convention.
+- The version used for the next release is determined by the highest version bump of any individual contract in the release.
+  - Example 1: The monorepo is at `op-contracts/v1.5.0`. Clarifying comments are made in contracts, so all contracts only bump the patch version. The next monorepo release will be `op-contracts/v1.5.1`.
+  - Example 2: The monorepo is at `op-contracts/v1.5.1`. Various tech debt and code is cleaned up in contracts, but no features are added, so at most, contracts bumped the minor version. The next monorepo release will be `op-contracts/v1.6.0`.
+  - Example 3: The monorepo is at `op-contracts/v1.5.1`. Legacy `ALL_CAPS()` getter methods are removed from a contract, causing that contract to bump the major version. The next monorepo release will be `op-contracts/v2.0.0`.
+- Feature specific monorepo releases (such as a beta release of the custom gas token feature) are supported, and should follow the guidelines in the [Smart Contract Feature Development](https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md) design doc. Bump the overall monorepo semver as required by the above rules, and append the `-beta,n` modifier to the version number. For example, if the last release before the custom gas token feature was `op-contracts/v1.5.1`, because the custom gas token introduces breaking changes, its beta release will be `op-contracts/v2.0.0-beta.1`.
+  - A subsequent release of the custom gas token feature that fixes bugs and introduces an additional breaking change would be `op-contracts/v2.0.0-beta.2`.
+  - This means `+feature-name` naming is not used for monorepo releases, only for individual contracts as described below.
+- A monorepo contracts release must map to an exact set of contract semvers, and this mapping must be defined in the contract release notes which are the source of truth. See [`op-contracts/v1.4.0-rc.4`](https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.4.0-rc.4) for an example of what release notes should look like.
+
+## Release Process
```

Comment:
Proposed change from @ajsutton:
- Develop on `develop` with `-rc.1` version
- Create release branch which removes the `-rc` qualifier and create a tag for governance off that branch
- When we want to put out an `.rc.2`, first make the change with a PR into develop (assuming it's feasible, which it should be if following https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md
- Then merge the updated develop into the release branch.

This ensures develop stays up to date during the governance process, and mitigates the risk of forgetting to merge the release back into the develop branch

Reply:
Incorporated in 3df49ea